From 521654f75f0ab6a5c11314edb9c12644b37772b3 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:41 -0400
Subject: [virt] kvm/nvmx: Advertise support for interrupt acknowledgement

Message-id: <1406040016-3289-187-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86265
O-Subject: [RHEL7 PATCH v2 186/221] KVM: nVMX: Advertise support for interrupt acknowledgement
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Bandan Das <bsd@redhat.com>

Some Type 1 hypervisors such as XEN won't enable VMX without it present

Signed-off-by: Bandan Das <bsd@redhat.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
(cherry picked from commit e0ba1a6ffcfe8dc95586943bbe56badb1459bf25)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 4378014..72b8012 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -2283,7 +2283,7 @@ static __init void nested_vmx_setup_ctls_msrs(void)
  rdmsr(MSR_IA32_VMX_EXIT_CTLS,
   nested_vmx_exit_ctls_low, nested_vmx_exit_ctls_high);
  nested_vmx_exit_ctls_low = VM_EXIT_ALWAYSON_WITHOUT_TRUE_MSR;
- /* Note that guest use of VM_EXIT_ACK_INTR_ON_EXIT is not supported. */
+
  nested_vmx_exit_ctls_high &=
 #ifdef CONFIG_X86_64
   VM_EXIT_HOST_ADDR_SPACE_SIZE |
@@ -2291,7 +2291,8 @@ static __init void nested_vmx_setup_ctls_msrs(void)
   VM_EXIT_LOAD_IA32_PAT | VM_EXIT_SAVE_IA32_PAT;
  nested_vmx_exit_ctls_high |= VM_EXIT_ALWAYSON_WITHOUT_TRUE_MSR |
   VM_EXIT_LOAD_IA32_EFER | VM_EXIT_SAVE_IA32_EFER |
-  VM_EXIT_SAVE_VMX_PREEMPTION_TIMER;
+  VM_EXIT_SAVE_VMX_PREEMPTION_TIMER | VM_EXIT_ACK_INTR_ON_EXIT;
+
  if (vmx_mpx_supported())
   nested_vmx_exit_ctls_high |= VM_EXIT_CLEAR_BNDCFGS;
 
-- 
1.7.1